rskj-core/src/main/java/co/rsk/db/importer/StateImporter.java

```diff
@@ -0,0 +1,79 @@
+package co.rsk.db.importer;
+
+import co.rsk.RskContext;
+import co.rsk.trie.Trie;
+import co.rsk.trie.TrieStore;
+import org.ethereum.core.Block;
+import org.ethereum.db.BlockStore;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Iterator;
+import java.util.Objects;
+
+public class StateImporter {
+
+    private final BlockStore blockStore;
+    private final TrieStore trieStore;
+
+    private StateImporter(RskContext ctx) {
+        this.blockStore = ctx.getBlockStore();
+        this.trieStore = ctx.getTrieStore();
+    }
+
+    public static void importDataIfRequired(RskContext ctx) {
+        if (ctx.getRskSystemProperties().databaseReset() || !ctx.getRskSystemProperties().databaseImport()) {
+            return;
+        }
+
+        String databaseDir = ctx.getRskSystemProperties().databaseDir();
+
+        StateImporter importer = new StateImporter(ctx);
+        URL bootstrapURL = getBootstrapURL(ctx);
+        BootstrapDataProvider bootstrapDataProvider = new BootstrapDataProvider(databaseDir, bootstrapURL, ctx);
+        importer.updateDatabaseIfNeeded(bootstrapDataProvider.getBootstrapBlockstore(), bootstrapDataProvider.getBootstrapTrieStore());
+    }
+
+    private static URL getBootstrapURL(RskContext ctx) {
+        String url = ctx.getRskSystemProperties().databaseImportUrl();
+        try {
+            return new URL(url);
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException(e);
```

Comment:
what's the plan for this exception? it's not caught as far as I can see

Reply:
It won't be catch. If the configuration is wrong the program will end. I'll improve the message for this :+1: